homeassistant/components/mqtt/__init__.py

```diff
@@ -34,6 +31,10 @@
     async_reload_integration_platforms,
 )
 from homeassistant.helpers.service import async_register_admin_service
+
+# MqttServiceInfo only for backwards compatibility, do not use this
+# as you'll get the whole MQTT integration via import.
+from homeassistant.helpers.service_info.mqtt import MqttServiceInfo  # noqa: F401
```

Comment:
What needs this backwards compatibility?

Reply:
Possibly custom integrations.

Reply:
> Should this line be adjusted, in `mqtt/discovery.py`, to load from the new helper?

That would be a bit cleaner.   I'll do that after breakfast.

Reply:
I couldn't find any with some basic searching but since we aren't targeting beta with this anymore, I think we take it out and if someone mentions its a problem we can decide how to handle it then.

I expect it is very unlikely its being used for custom integrations

Reply:
1be662a3da0c7cae1af80227c6dcaafb811099c0